src/components/tl/cuda/tl_cuda_nvls.c

```diff
@@ -385,6 +386,29 @@ ucc_status_t ucc_tl_cuda_nvls_init(struct ucc_tl_cuda_team *self,
     nvls->mc_memhandle = memhandle;
     nvls->mc_size      = mcSize;
     nvls->mc_offset    = mcOffset;
+    nvls->coll_ids     = ucc_malloc(lib->cfg.max_concurrent * sizeof(size_t), "coll_ids");
+    if (!nvls->coll_ids) {
+        status = UCC_ERR_NO_MEMORY;
+        goto cleanup;
+    }
+    for (i = 0; i < lib->cfg.max_concurrent; ++i) {
+        nvls->coll_ids[i] = 0;
+    }
+
+    if (UCC_TL_TEAM_RANK(self) == 0) {
+        // root rank initializes the arrival counter for each task
+        ucc_tl_cuda_nvls_control_t control;
+        control.arrival_counter = 0;
+
+        for (i = 0; i < lib->cfg.max_concurrent; ++i) {
+            void *control_uc =
```

Comment:
can we use TASK_NVLS_CONTROL_UC here?

Reply:
No, since there is no tasks here. This is team initialization stage.